src/sim/include/AbstractDriver.h

```diff
@@ -1247,21 +1220,31 @@ namespace MFM
 	    //todo: figure out why that doesn't work. GetAtomInSite has an option to get the base layer, which would
 	    //be nice to have. 
             T* atom = grid.GetWritableAtom(siteInGrid);
+	    const T empty = tile.GetEmptyAtom();
+	    const u32 t = atom->GetType();
+	    if(atom->IsType(empty, t)) continue;
 
             UlamClassRegistry<EC> ucr = grid.GetUlamClassRegistry();
 
-	    const u32 t = atom->GetType();
 	    const Element<EC> * e = grid.LookupElement(t);
             const UlamElement<EC> * uelt = e->AsUlamElement();
 
+	    //todo: are there any alternatives to the string un this buffer that contain the names of the data members and their value at current epoch?
```

Comment:
I'd consider adding like 
PRINT_FORMAT_JSON = 0x00001000, //< Format data members in JSON
at UlamClass.h:71 or so (hmm and at DebugUtils.ulam:20 or so in ULAM),
then checking for PRINT_FORMAT_JSON like around UlamClass.tcc:103 and 110.
Not super easy but simplifies and flattens the JSON

Reply:
If it drops the need for anyone else to have to use the Python deserializer I wrote or otherwise implement their own, I think it's worth it. I'll see if I can pick up on what you're suggesting when I get to this tonight. 
I assume the idea would be to add an additional method in order to not mess up the display of data member values in the mfms gui; is that the right way?

Reply:
I think the new commits are in the spirit of this. I added this flag and added a condition for this in UlamClass.tcc. It could be prettier since I repeated most of what's already there. I can clean this up, but the resulting output looks closer to what I was hoping for in the first place.